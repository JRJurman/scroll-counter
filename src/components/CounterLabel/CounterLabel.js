import { registerHtml, useState } from 'tram-one'
import './CounterLabel.scss'
const html = registerHtml()

export default () => {
  const [label, setLabel] = useState('')
  const onChangeInput = (event) => setLabel(event.target.value)
  return html`
    <input class="CounterLabel" placeholder="Counter" value=${label} onchange=${onChangeInput} />
  `
}
