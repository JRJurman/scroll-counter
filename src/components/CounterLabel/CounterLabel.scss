.CounterLabel {
  background: inherit;
  border: none;
  outline: none;
  color: inherit;
  width: 90%;
  font-size: 1em;
  font-weight: bold;
  padding: 0 5%;
}

.CounterLabel::-webkit-input-placeholder { /* Chrome/Opera/Safari */
  color: rgba(0, 0, 0, 0.5);
}
.CounterLabel::-moz-placeholder { /* Firefox 19+ */
  color: rgba(0, 0, 0, 0.5);
}
.CounterLabel:-ms-input-placeholder { /* IE 10+ */
  color: rgba(0, 0, 0, 0.5);
}
.CounterLabel:-moz-placeholder { /* Firefox 18- */
  color: rgba(0, 0, 0, 0.5);
}
