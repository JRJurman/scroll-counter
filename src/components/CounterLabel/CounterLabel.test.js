import CounterLabel from './CounterLabel'

describe('CounterLabel', () => {
  describe('with default props', () => {
    const wrapper = CounterLabel()

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
