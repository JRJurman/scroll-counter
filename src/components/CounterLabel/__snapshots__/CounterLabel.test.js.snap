// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterLabel with default props should match snapshot 1`] = `"<input placeholder=\\"Counter\\" value=\\"\\" class=\\"CounterLabel\\">"`;
